packages/pigeon/lib/functional.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// A [map] function that calls the function with an enumeration.
```

Comment:
I can't tell from this description what this does. Looking at the code, it appears that it's an indexed map? It needs a description that explains that. (And if the int is the index of the enumeration, why not call this `indexedMap` or `mapWithIndex`, and change `x` to `index`?)

Reply:
Sorry, this isn't required, removed.  It was accidentally introduced when I was shuffling the misplaced changes.  This file should be introduced in a later PR.